lib/bolt/config/transport/ssh.rb

```diff
@@ -34,6 +34,7 @@ class SSH < Base
 
         # Options available when using the native ssh transport
         NATIVE_OPTIONS = %w[
+          batch-mode
```

Comment:
Since this will *always* hang when `ssh-command` isn't set (since it's not something you can set outside of Bolt config), it seems safe to error (or at a minimum warn) when batch-mode is disabled and `ssh-command` isn't configured.

Reply:
Oo, ok, I'd definitely lean towards erroring, since at least for me this is what happens when I set `batch-mode: false` without an `ssh-command`:
```
lucy ~/githubs/bolt ((3b0135e8...)) $ be bolt command run whoami -t docker_ssh
Started on 127.0.0.1:20022...
Started on 127.0.0.1:20024...
Started on 127.0.0.1:20025...
bolt@127.0.0.1's password: 
bolt@127.0.0.1's password: 
bolt@127.0.0.1's password: 
bolt@127.0.0.1's password: 
bolt@127.0.0.1's password: 
bolt@127.0.0.1's password: 
Failed on 127.0.0.1:20022:
  Failed to connect to 127.0.0.1:20022: Permission denied, please try again.
  Permission denied, please try again.
  Received disconnect from 127.0.0.1 port 20022:2: Too many authentication failures
  Disconnected from 127.0.0.1 port 20022


bolt@127.0.0.1's password: 
bolt@127.0.0.1's password: 
bolt@127.0.0.1's password: 
bolt@127.0.0.1's password: 
Failed on 127.0.0.1:20025:
  The command failed with exit code 255
  Permission denied, please try again.
  Permission denied, please try again.
  Received disconnect from 127.0.0.1 port 20025:2: Too many authentication failures
  Disconnected from 127.0.0.1 port 20025
  

bolt@127.0.0.1's password: 
Failed on 127.0.0.1:20024:
  The command failed with exit code 255
  Permission denied, please try again.
  Permission denied, please try again.
  Received disconnect from 127.0.0.1 port 20024:2: Too many authentication failures
  Disconnected from 127.0.0.1 port 20024
Failed on 3 targets: 127.0.0.1:20022,127.0.0.1:20024,127.0.0.1:20025
Ran on 3 targets in 15.63 sec
```